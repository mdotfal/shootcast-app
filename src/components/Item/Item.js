import React, { Component } from 'react';

class Item extends Component {

  state = {
    city: "",
  }

  getNameOnClick = e => {
    console.log( 'getNameOnClick clicked', this.props.city.name )
    this.setState({
      city: this.props.city.name
    })
  }

  render() {
    // console.log( 'Item-this.state',this.state )
    return (
      <div className="item">
          <div 
            className="city-item"
            onClick={ e => this.getNameOnClick( e ) }>
              
             { this.props.city.name }

            <button
              className="btn"
              type="button"
              onClick={ () => this.props.onDeleteCity( this.props.city )}
            >x</button>
          </div>
      </div>
    );

  }
};

export default Item;