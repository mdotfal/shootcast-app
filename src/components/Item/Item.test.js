import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Item from './Item';

describe( `Item component`, () => {
  
  const params = {
    city: {
      name: "San Francisco"
    }
  }

  it( `renders without crashing`, () => {
    const div = document.createElement( 'div' );
    ReactDOM.render( <Item />, div );
    ReactDOM.unmountComponentAtNode( div );
  })

  it(Â `renders with params`, () => {
    const wrapper = shallow( <Item { ...params }/> )
    expect( toJson( wrapper )).toMatchSnapshot()
  })
})