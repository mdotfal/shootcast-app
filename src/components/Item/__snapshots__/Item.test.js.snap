// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component renders with params 1`] = `
<div
  className="item"
>
  <div
    className="city-item"
    onClick={[Function]}
  >
    San Francisco
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>
</div>
`;
